wordpress-mobile/WordPressKit-iOS
# Remove AFNetworking part 2: Refactor WordPressComRestAPI

This PR updates the WordPressComRestAPI to use AlamoFire instead of AFNetworking.

On the way I move the check for calls failing because of too many requests to the inside of the API object instead of being done on the Remote object.

How to test:
 - Make sure the unit tests are passing.
 - Test on the main app using this branch: `try/remove_af_networking_from_wpkit`
 - Because of the amount of places this code is used check several parts of the app:
  -  Post and page lists;
  -  Reader;
  - Site List;
  - Notifications;
  - Editor
  - Media Library, upload multiple images and videos and check memory impact
  - See if there is any slow downs while scrolling around
 - Test on a device!
